lib/failure_detector.ml

```diff
@@ -0,0 +1,131 @@
+open Base.Hashtbl
+
+type message =
+  | Ping
+  | Acknowledge
+  | PingRequest of Address.t
+[@@deriving bin_io]
+
+type config = {
+  protocol_period : int;
+  round_trip_time : int;
+  timeout : int;
+  mutable peers_to_ping : int;
+}
+
+type t = {
+  config : config;
+  (* TODO: I am totally not sure about this type *)
+  acknowledges : (int, unit Lwt_condition.t) Base.Hashtbl.t;
```

Comment:
Just a comment to don't lose the conversation about this type holding something like (sequence, status) for each peer.

Reply:
Hmm, I think I understand better what this is now.